### PR TITLE
cache: remove buffer_acquire/release part 2

### PR DESCRIPTION
**Pull request by tobonex**
This is step 4 of implementation of #8006
This PR is a result of splitting https://github.com/thesofproject/sof/pull/8106 into parts.

Remove usage of buffer_acquire and buffer_release from a number of files.

next part: https://github.com/thesofproject/sof/pull/8218

### CODE REVIEW

**Review by btian1 — APPROVED** [review]

File: src/audio/smart_amp/smart_amp.c [hunk 1]
```diff
@@ -589,56 +582,49 @@ static int smart_amp_copy(struct comp_dev *dev)
...
 	/* available bytes and samples calculation */
-	avail_passthrough_frames = audio_stream_avail_frames(&source_buf->stream,
-							     &sink_buf->stream);
+	avail_passthrough_frames = audio_stream_avail_frames(&sad->source_buf->stream,
+							     &sad->sink_buf->stream);
 
 	avail_frames = avail_passthrough_frames;
 
 	if (sad->feedback_buf) {
-		struct comp_buffer *feedback_buf = buffer_acquire(sad->feedback_buf);
```
Comment (lyakh) [inline src/audio/smart_amp/smart_amp.c:598]: maybe we should keep this pointer - just remove the call to `buffer_acquire()`. This is a `.copy()` function and our recent trend has been distrusting the compiler to make simple meaningful optimisations like this one and doing them manually... Maybe for `source_buf` and `sink_buf` too. That also reduces the chance of an accidental mistake

**Review by kv2019i — CHANGES_REQUESTED** [review]
Thanks @tobonex , one question inline, rest seems good.

File: src/audio/copier/copier.c [hunk 2]
```diff
@@ -543,24 +532,16 @@ static int copier_multi_endpoint_dai_copy(struct copier_data *cd, struct comp_de
...
-	src_c = buffer_acquire(src);
 
 	/* gateway(s) on output */
-	sink_c = buffer_acquire(cd->multi_endpoint_buffer);
-	ret = do_conversion_copy(dev, cd, src_c, sink_c, &processed_data);
-	buffer_release(sink_c);
-
-	if (ret < 0)
-		goto err;
+	ret = do_conversion_copy(dev, cd, src, cd->multi_endpoint_buffer, &processed_data);
```
[the remark this answers is not in the document]
Comment (tobonex) [reply src/audio/copier/copier.c:540]: Huh, missed it somehow. Fixed it now.

File: src/audio/smart_amp/smart_amp.c (hunk 1 above)
Comment (tobonex) [reply src/audio/smart_amp/smart_amp.c:598]: Does it look good now?

**Review by lgirdwood — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
Thanks @tobonex , looks good now!

**Comment by kv2019i** [comment]
This is ready to go, only needs a clean run for mandatory CI checks. We had a momentary hickup today, so re-push will probably make the PR pass now. I'd prefer we'll merge these acquire/release PRs at a one-per-day rate, so we can get some additional feedback from daily test rounds and will then have less things to revert in case we miss something.

**Review by lyakh — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Comment by tobonex** [comment]
just re-pushing

**Review by marcinszkudlinski — APPROVED** [review]

**Comment by kv2019i** [comment]
The alsabat tests are failing on multiple nocodec tplg platforms due to https://github.com/thesofproject/sof/issues/8238 , rest of tests are good. One Windows build failed due to missing Ninja tool, but other Windows builds go through so this seems transient and anyways no connection to code changes in this PR. So with that, proceeding with this. @tobonex , I'll target to merge part 3 tomorrow, so we'll get some daily test feedback for part 2.

**Pull request merged**